gusto/core/io.py

```diff
@@ -761,74 +761,92 @@ def dump(self, state_fields, time_data):
-
-            # Add coordinates if they are not already in the file
-            for space_name in space_names:
-                if space_name not in self.domain.coords.chi_coords.keys():
-                    # Space not registered
-                    # TODO: we should fail here, but currently there are some spaces
-                    # that we can't output for so instead just skip outputting
-                    pass
+                # At present parallel netCDF crashes when saving strings. To get around this
+                # we instead save string metadata as char arrays of fixed length.
+                if isinstance(output_value, str):
+                    nc_field_file.createVariable(metadata_key, 'S1', ('dim_one', 'dim_string'))
+                    output_char_array = np.array([output_value], dtype='S256')
+                    nc_field_file[metadata_key][:] = stringtochar(output_char_array)
                 else:
-                    coord_fields = self.domain.coords.global_chi_coords[space_name]
-                    num_points = len(self.domain.coords.global_chi_coords[space_name][0])
-
-                    nc_field_file.createDimension('coords_'+space_name, num_points)
-
-                    for (coord_name, coord_field) in zip(self.domain.coords.coords_name, coord_fields):
-                        nc_field_file.createVariable(coord_name+'_'+space_name, float, 'coords_'+space_name)
-                        nc_field_file.variables[coord_name+'_'+space_name][:] = coord_field[:]
-
-            # Create variable for storing the field values
-            for field in self.to_dump:
-                field_name = field.name()
-                space_name = field.function_space().name
-                if space_name not in self.domain.coords.chi_coords.keys():
-                    # Space not registered
-                    # TODO: we should fail here, but currently there are some spaces
-                    # that we can't output for so instead just skip outputting
-                    logger.warning(f'netCDF outputting for space {space_name} '
-                                   + 'not yet implemented, so unable to output '
-                                   + f'{field_name} field')
-                else:
-                    nc_field_file.createGroup(field_name)
-                    nc_field_file[field_name].createVariable('field_values', float, ('coords_'+space_name, 'time'))
+                    nc_field_file.createVariable(metadata_key, type(output_value), ('dim_one',))
+                    nc_field_file[metadata_key][0] = output_value
+
+        # Add coordinates if they are not already in the file
+        for space_name in space_names:
+            if space_name not in self.domain.coords.chi_coords.keys():
+                # Space not registered
+                # TODO: we should fail here, but currently there are some spaces
+                # that we can't output for so instead just skip outputting
+                pass
```

Comment:
Do we need to keep the logged warning here?

Reply:
It's unclear but this is actually what was already there on line 791. I can amend if needed.